packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -220,6 +230,108 @@ function getDefaultEnv(context: $TSContext): string | undefined {
   return undefined;
 }
 
+async function setEnvVar(context: $TSContext, currentEnvName: string, projectPath: string) {
```

Comment:
all of the env var logic needs to be contained in the functions category package. The `amplify-cli` package shouldn't have knowledge of function env vars. The platform will call [this function](https://github.com/aws-amplify/amplify-cli/blob/23fcfa532b53366fd4d371f4913806cd078f723e/packages/amplify-category-function/src/index.ts#L105) on env initialization which is where the entry point to this logic should be located

Reply:
@edwardfoyle We may want to keep having the logic as a part of init-steps because we cannot add a question to beginning of `env add` if we move the logic to `amplify-category-function`. `isNewEnv` is called after CFN workflow:

<img width="292" alt="Screen Shot 2021-05-25 at 18 17 11" src="https://user-images.githubusercontent.com/1716463/119472731-7370bb80-bd85-11eb-8f01-af58752d721a.png">

Reply:
I moved the logic into `amplify-category-function` :)